### ISSUE TITLE
(🐞) false error, argument to `__call__`  with conditional expression

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#16033]
```py
from typing import Callable, TypeVar

T = TypeVar("T")

class Class:
    def __call__(self, fn: Callable[[], T]) -> Callable[[], T]:
        return fn

def func(fn: Callable[[], None]):
    Class()(fn)  # no error
    Class() if bool() else Class()(fn)  # error: Argument 1 to "__call__" of "Class" has incompatible type "Callable[[], None]"; expected "Callable[[], Callable[[], T]]"  [arg-type]
```

**Comment by ilevkivskyi** [comment]
This example now works correctly.
